pkg/api/v1/conversion.go

```diff
@@ -290,7 +282,14 @@ func Convert_v1_PodStatusResult_To_api_PodStatusResult(in *PodStatusResult, out
 	if err := autoConvert_v1_PodStatusResult_To_api_PodStatusResult(in, out, s); err != nil {
 		return err
 	}
-	delete(out.Annotations, PodInitContainerStatusesAnnotationKey)
+	if len(out.Annotations) > 0 {
```

Comment:
This block is deleting the status annotation - if there are no annotations, there's nothing to delete and nothing to copy.